### PR TITLE
feature: add image filter option dangling

### PR DESCRIPTION
**Pull request by lang710**
Signed-off-by: Lang Chi <21860405@zju.edu.cn>

### Ⅰ. Describe what this PR did
Add the dangling filter option for pouch images.

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
Add it later.

### Ⅳ. Describe how to verify it
```
# pouch images -f "dangling=true"
# pouch images -f "dangling=false"
# pouch images --filter "dangling=true"
# pouch images --filter "dangling=false"
```

### CODE REVIEW

File: daemon/mgr/image.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ var acceptedImageFilterTags = map[string]bool{
 	"before":    true,
 	"since":     true,
 	"reference": true,
+	"dangling":  true,
```
Comment (ZYecho) [inline daemon/mgr/image.go]: we also need to change related docs, such as swagger or cmd desc.

File: daemon/mgr/image.go [hunk 2]
```diff
@@ -299,6 +308,21 @@ func (mgr *ImageManager) ListImages(ctx context.Context, filter filters.Args) ([
 			continue
 		}
 
+		if len(danglingImages) > 0 {
+			danglingFilter := danglingImages[0]
+			if danglingFilter == trueString {
+				if imgInfo.RepoDigests != nil || imgInfo.RepoTags != nil {
```
Comment (ZYecho) [inline daemon/mgr/image.go]: do we need to judge that `imgInfo.RepoDigests` or `imgInfo.RepoTags` is a empty slice?

File: daemon/mgr/image.go [hunk 3]
```diff
@@ -41,8 +41,12 @@ var acceptedImageFilterTags = map[string]bool{
 	"before":    true,
 	"since":     true,
 	"reference": true,
+	"dangling":  true,
 }
 
+const trueString = "true"
```
Comment (ZYecho) [inline daemon/mgr/image.go]: why need this?

File: daemon/mgr/network.go [hunk 4]
```diff
@@ -528,11 +528,11 @@ func networkOptions(create apitypes.NetworkCreateConfig) ([]libnetwork.NetworkOp
 	}
 
 	// parse options
-	if v, ok := networkCreate.Options["persist"]; ok && v == "true" {
+	if v, ok := networkCreate.Options["persist"]; ok && v == trueString {
 		nwOptions = append(nwOptions, libnetwork.NetworkOptionPersist(true))
 		delete(networkCreate.Options, "persist")
 	}
-	if v, ok := networkCreate.Options["dynamic"]; ok && v == "true" {
```
Comment (ZYecho) [inline daemon/mgr/network.go]: ?

**Comment by ZYecho** [comment]
could you add a test case for this feature?

File: daemon/mgr/image.go (hunk 1 above)
Comment (lang710) [reply daemon/mgr/image.go]: Yes.

File: daemon/mgr/network.go (hunk 4 above)
Comment (lang710) [reply daemon/mgr/network.go]: In the "mgr" package, "true" appears three times, so I change it to a constant.

**Comment by fuweid** [comment]
how to define dangling? I don't think pouch container needs this.

File: daemon/mgr/network.go (hunk 4 above)
Comment (ZYecho) [reply daemon/mgr/network.go]: but in my opnion this change didn't make code more readable.

Comment (lang710) [reply daemon/mgr/network.go]: Yeah, I'll take your advise.

**Pull request closed without merging**